ccmgradm/main.go

```diff
@@ -202,6 +224,94 @@ func main() {
 				os.Exit(1)
 			}
 		}
+	case args.Init != nil:
+		{
+			saveAndReload = false
+			if args.Init.LocalCmon {
+				cmonUrl := "127.0.0.1:9501"
+				if len(args.Init.Url) > 0 {
+					cmonUrl = args.Init.Url
+				}
+				cmon := cfg.ControllerByUrl(cmonUrl)
+				if cmon != nil {
+					fmt.Println("Controller already exists with this URL.")
+					os.Exit(1)
+				}
+				cmon = &config.CmonInstance{
+					Xid:         xid.New().String(),
+					Url:         cmonUrl,
+					Name:        "local",
+					UseCmonAuth: true,
+					FrontendUrl: "localhost",
+				}
+
+				// configFile := "/etc/cmon.cnf"
+				var cmonConfig string
+				var cfgFile *ini.File
+				if len(args.Init.Config) > 0 {
+					cmonConfig = args.Init.Config
+				}
+				if cmonConfig != "" {
+					cfgFile, err = ini.Load(configFile)
+					if err != nil {
+						fmt.Println("Error loading cmon.cnf file:", err.Error())
+						os.Exit(1)
+					}
+					rpcKey := cfgFile.Section("").Key("rpc_key").String()
+					if rpcKey == "" {
+						fmt.Println("Error, rpc_key not found in .cnf file.")
+						os.Exit(1)
+					}
+					cmon.Username = "ccrpc"
+					cmon.Password = rpcKey
+				}
+
+				if err := cfg.AddController(cmon, false); err != nil {
+					fmt.Println("Couldn't add controller:", err.Error())
+					os.Exit(1)
+				}
+
+				fmt.Println("Controller", cmonUrl, "registered successfully")
+
+			}
+			if args.Init.Port > 0 && cfg.Port != args.Init.Port {
+				fmt.Println("Changing port from", cfg.Port, "to", args.Init.Port)
+				cfg.Port = args.Init.Port
+			}
+			if len(args.Init.FrontendPath) > 0 && cfg.FrontendPath != args.Init.FrontendPath {
+				fmt.Println("Changing frontend_path from", cfg.FrontendPath, "to", args.Init.FrontendPath)
+				cfg.FrontendPath = args.Init.FrontendPath
+				func() {
+					// @todo get rid of config.js file and generate env variables for UI dynamically
+					filePath := fmt.Sprintf("%s/config.js", cfg.FrontendPath)
+					input, err := os.ReadFile(filePath)
+					if err != nil {
+						fmt.Println(err)
+						return
+					}
+					fileInfo, err := os.Stat(filePath)
+					if err != nil {
+						fmt.Println(err)
+						return
+					}
+					content := string(input)
+					updatedContent := strings.Replace(content, "CMON_API_URL: '/api/v2'", "CMON_API_URL: '/'", 1)
+					err = os.WriteFile(filePath, []byte(updatedContent), fileInfo.Mode().Perm())
```

Comment:
this would not work in k8s most likely

Reply:
i left todo there to generate this config dynamically, perhaps can be used env variable there. Would that cover k8s case? i would take in a separate ticket